### PR TITLE
Integrate format code script

### PR DESCRIPTION
**Pull request by Alan-Jowett**
This pull request introduces two new scripts, `format-code` and `format-code.ps1`, to format C/C++ source files based on `.clang-format` rules. Additionally, it updates the `pre-commit` script to include a formatting check. Here are the most important changes:

### New Formatting Scripts:

* [`scripts/format-code`](diffhunk://#diff-f72bea0785fb38e7f6fdb0bcf18e46e51d3eb05a6c5538b9d2c0cdf1f5111573R1-R304): Added a Bash script to format C/C++ source files. It includes options for verbose output, quiet mode, staged files, and what-if scenarios. It also checks for an acceptable version of `clang-format` and processes command-line options.
* [`scripts/format-code.ps1`](diffhunk://#diff-4b6da8ee05099f3f0e5d67e9f366f6ac54c3ecfd8ba16d85e50f204056e09fd0R1-R337): Added a PowerShell script with similar functionality to the Bash script, including handling of command-line options and checking for `clang-format`. It also includes verbose and what-if modes.

### Pre-commit Hook Update:

* [`scripts/pre-commit`](diffhunk://#diff-40564a4a456c31b8543923ab58e31689dde6119d9419c1211a0cbee86a524debR40-R43): Updated to include a call to the new `format-code` script to check for formatting issues before committing. If formatting issues are found, the commit is aborted with instructions to fix the formatting.

## Summary by CodeRabbit

- **New Features**
	- Introduced `format-code` scripts for formatting C/C++ source files according to `.clang-format` rules, available in both Bash and PowerShell.
	- Added a pre-commit hook to validate code formatting before commits, ensuring adherence to formatting standards.

- **Bug Fixes**
	- Enhanced error handling in the formatting scripts to manage failures and unknown options effectively.

### CODE REVIEW

**Review by elazarg — COMMENTED** [review]
A couple of questions:
1. Why specifically release 17.0.3? I now see that `BreakAfterAttributes: Always` was introduced in v16.
2. Can you check that the version is at least required_cfver, not equivalent to it?
3. How about using [git-clang-format](https://clang.llvm.org/docs/ClangFormat.html#git-integration)? I've just find out about it, and it has the benefit of not bloating the diff when some previous commit was not properly formatted.

**Comment by Alan-Jowett** [comment]
> A couple of questions:
> 
> 1. Why specifically release 17.0.3? I now see that `BreakAfterAttributes: Always` was introduced in v16.
> 2. Can you check that the version is at least required_cfver, not equivalent to it?
> 3. How about using [git-clang-format](https://clang.llvm.org/docs/ClangFormat.html#git-integration)? I've just find out about it, and it has the benefit of not bloating the diff when some previous commit was not properly formatted.

This code was taken from eBPF for Windows repo, so may need some tweaks.

17.0.3 was selected as that's the version that eBPF for Windows mandates (or rather that's the version that ships with Visual Studio and is compiled by MSFT and hence is considered safer from supply chain attacks).

Follow-ups:
1) Make this a minimum version.
2) Investigate git-clang-format.

Note:
This script only checks changes files, but will reformat prior commits in the current files.

**Review by elazarg — APPROVED** [review]

File: scripts/format-code [hunk 1]
```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env bash
+
+# Copyright (c) eBPF for Windows contributors
```
Comment (dthaler) [inline scripts/format-code:3]:
```suggestion
# Copyright (c) Prevail Verifier contributors.
```

File: scripts/format-code.ps1 [hunk 2]
```diff
@@ -0,0 +1,337 @@
+# Copyright (c) eBPF for Windows contributors
```
Comment (dthaler) [inline scripts/format-code.ps1:1]:
```suggestion
# Copyright (c) Prevail Verifier contributors.
```

**Review by dthaler — APPROVED** [review]

**Pull request merged**